File: frontend/src/types/video.ts
export interface Video {
  id: number
  url: string
  youtube_id: string
  title: string
  description?: string
  thumbnail_url?: string
  user_email?: string
  user_id: number
  created_at: string
  updated_at: string
}

export interface Pagination {
  current_page: number
  total_pages: number
  total_count: number
  per_page: number
  next_page: number | null
  prev_page: number | null
}

export interface VideoListResponse {
  success: boolean
  videos: Video[]
  count: number
  pagination: Pagination
}

File: backend/app/models/video.rb
class Video < ApplicationRecord
  belongs_to :user
  validates :url, presence: true
  validates :youtube_id, presence: true,
                         uniqueness: {
                           scope: :user_id,
                           message: "You have already shared this video."
                         }
  validates :title, presence: true

  has_many :bookmarks, dependent: :destroy

  def self.ransackable_attributes(_auth_object = nil)
    %w[created_at description id thumbnail_url title updated_at url user_id youtube_id]
  end

  def self.ransackable_associations(_auth_object = nil)
    %w[bookmarks user]
  end
end
